core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -122,6 +123,24 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
       val externalAccumulables = allAccumulables.values.filter { acc => !acc.internal }
       val hasAccumulators = externalAccumulables.size > 0
 
+      val tasksLocality = new mutable.HashMap[String, Long]
+      if (stageData.taskData.size > 0) {
+        for ((key, value) <- stageData.taskData) {
+          val locality = value.taskInfo.taskLocality.toString;
```

Comment:
@kayousterhout  Thanks so much for all your feedbacks !!! Much appreciated. Let me update the PR accordingly.